src/infill/LightningLayer.cpp

```diff
@@ -260,7 +260,15 @@ Polygons LightningLayer::convertToLines(const coord_t line_width) const
 
     for (const auto& tree : tree_roots)
     {
-        tree->convertToPolylines(result_lines, line_width);
+        // If even the furthest location in the tree is inside the polygon, the entire tree must be inside of the polygon.
+        // (Don't take the root as that may be on the edge and cause rounding errors to register as 'outside'.)
+        constexpr coord_t epsilon = 5;
+        Point should_be_inside = tree->getLocation();
+        PolygonUtils::moveInside(limit_to_outline, should_be_inside, epsilon, epsilon * epsilon);
+        if (limit_to_outline.inside(should_be_inside))
```

Comment:
Effectively, `moveInside` already checks if it's inside. This extra check (`O(N)`) is unnecessary if we look at how far the `moveInside` needs to move the point instead of relying on the maximum move distance:

```suggestion
        PolygonUtils::moveInside(limit_to_outline, should_be_inside);
        if (vSize2(tree->getLocation() - should_be_inside) < epsilon * epsilon)
```

We decided however that since this is only a performance issue, we move that out of scope for now and make that improvement to Master, and merge this for 4.12.